app/config/MicroserviceAppConfig.scala

```diff
@@ -56,12 +58,31 @@ class MicroserviceAppConfig @Inject()(servicesConfig: ServicesConfig) {
     )
   }
 
+  val hipUrl: String = loadConfig("microservice.services.hip.url")
+  def getHipCredentials(api: String): String = {
+    val clientId = loadConfig(s"microservice.services.hip.$api.clientId")
+    val secret = loadConfig(s"microservice.services.hip.$api.secret")
+
+    val encoded = Base64.getEncoder.encodeToString(s"$clientId:$secret".getBytes("UTF-8"))
+
+    s"Basic $encoded"
+  }
+
+  def getHIPHeaders(api: String, requestId: Option[RequestId]): Seq[(String, String)] = {
+    Seq(
+      (HeaderNames.authorisation, getHipCredentials(api)),
+      ("correlationId", requestId.map(_.value).getOrElse(UUID.randomUUID().toString))
+    )
+  }
+
+  def isHIPFeatureSwitchEnabled(api: String): Boolean = {
```

Comment:
can this be renamed `hipFeatureSwitchEnabled`? so when used  it would read something like this:

```
If(hipFeatureSwitchEnabled) { } else {} 
```
or 
```
case  _ => if(hipFeatureSwitchEnabled)
```

Reply:
sorry thinking now if we have more apis using enums here would be better to help prevent string errors. 

each enum will have a def name() of the api defined as a string so if we call the method multiple times for different apis we only have one place where the string will be set. 
 
```
def hipFeatureSwitchEnabled(api: SealedTraitEnumType): Boolean = {
    servicesConfig.getBoolean(s"microservice.services.hip.${api.name}.feature-switch")
}
```

Happy to leave it for now though.